### PR TITLE
(Ready for review) Support for beautiful relative imports

### PR DESCRIPTION
**Pull request by KevinHock**
This PR started as adding support for relative imports, but naturally became an inter-procedural PR including tests for things I haven't done yet like init files, `__all__`, `from foo import *` with style and  random improvements e.g. `project_handler_test` along the way.

A few of the small things:

- `get_python_modules` -> `get_modules` because I added `get_modules_and_packages`

- `is_python_module -> is_python_file`

- `add all tuples to project handler test`

- `double-quotes all around, for consistent multi-line comments`

### CODE REVIEW

**Comment by KevinHock** [comment]
I just realized I never handled 'from . import foo', I'll be sure to add a test for that.

**Comment by KevinHock** [comment]
The more I think about it, there are more cases I haven't handled, we don't parse init files yet, right?

File: pyt/base_cfg.py [hunk 1]
```diff
@@ -252,6 +252,7 @@ def connect_control_flow_node(self, control_flow_node, next_node):
     def connect_nodes(self, nodes):
         """Connect the nodes in a list linearly."""
         for n, next_node in zip(nodes, nodes[1:]):
+
```
Comment (KevinHock) [inline pyt/base_cfg.py]: extra line, I'll remove it

File: pyt/base_cfg.py [hunk 2]
```diff
@@ -286,7 +287,7 @@ def stmt_star_handler(self, stmts):
             elif isinstance(node, BreakNode):
                 break_nodes.append(node)
 
-            if self.node_to_connect(node):
+            if self.node_to_connect(node) and node:
```
Comment (KevinHock) [inline pyt/base_cfg.py]: connect_nodes messes up if node is None

File: pyt/interprocedural_cfg.py [hunk 3]
```diff
@@ -219,9 +219,12 @@ def save_actual_parameters_in_temp(self, args, arguments, line_number):
 
             label_visitor = LabelVisitor()
             label_visitor.visit(parameter)
+            rhs_visitor = RHSVisitor()
+            rhs_visitor.visit(parameter)
+
             n = RestoreNode(temp_name + ' = ' + label_visitor.result,
                             temp_name,
-                            [label_visitor.result],
+                            rhs_visitor.result,
```
Comment (KevinHock) [inline pyt/interprocedural_cfg.py]: This was the bug that messed up inter-procedural analysis.

File: pyt/interprocedural_cfg.py [hunk 4]
```diff
@@ -346,7 +349,15 @@ def visit_Call(self, node):
         self.function_return_stack.append(_id)
 
         local_definitions = self.module_definitions_stack[-1]
-        definition = local_definitions.get_definition(_id)
+
+        real_id = _id
+        for k in local_definitions.import_alias_mapping.keys():
+            if _id.startswith(k):
+                real_id = _id.replace(k, local_definitions.import_alias_mapping[k])
+                definition = local_definitions.get_definition(real_id)
```
Comment (KevinHock) [inline pyt/interprocedural_cfg.py]: This makes sure, where `import bar as foo` if it sees foo it gets the definition of bar.

File: tests/vulnerabilities_across_files_test.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        vulnerability_log = self.run_analysis('example/vulnerable_code_across_files/absolute_from_file_command_injection_2.py')
+        self.assert_length(vulnerability_log.vulnerabilities, expected_length=1)
+
+    def test_no_false_positive_absolute_from_file_command_injection_3(self):
+        vulnerability_log = self.run_analysis('example/vulnerable_code_across_files/no_false_positive_absolute_from_file_command_injection_3.py')
+        self.assert_length(vulnerability_log.vulnerabilities, expected_length=0)
+
+    # This fails due to a false positive in get_vulnerability
+    # def test_absolute_from_file_does_not_exist(self):
+    #     vulnerability_log = self.run_analysis('example/vulnerable_code_across_files/absolute_from_file_does_not_exist.py')
```
Comment (KevinHock) [inline tests/vulnerabilities_across_files_test.py]: So when a function has to be treated as a blackbox it shouldn't assume the return value is tainted, but it's somewhat low priority to fix, for me, personally.

File: tests/nested_functions_test.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# import os.path
+
+# from .base_test_case import BaseTestCase
+# from pyt.project_handler import get_directory_modules, get_modules_and_packages
+
+
+# class NestedTest(BaseTestCase):
+    # def test_nested_function_calls(self):
+
+    #     path = os.path.normpath('example/nested_functions_code/nested_function_calls.py')
```
Comment (KevinHock) [inline tests/nested_functions_test.py]: Not gonna fix it in this PR, I wanted to share the test though.

File: tests/import_test.py [hunk 7]
```diff
@@ -2,22 +2,395 @@
...
+                    "b = save_3_b",
+                    "¤call_3 = ret_duck_house",
+                    "c = ¤call_3",
+                    "Exit module"]
+
+
+        for node, expected_label in zip(self.cfg.nodes, EXPECTED):
+            self.assertEqual(node.label, expected_label)
+
+    # def test_init(self):
```
Comment (KevinHock) [inline tests/import_test.py]: As I started to try to get this to pass I thought the PR would be too big so I just pushed and wrote

```python
raise Exception("TODO: Handle __init__ files")
```

File: pyt/interprocedural_cfg.py (hunk 4 above)
Comment (Thalmann) [inline pyt/interprocedural_cfg.py]: Nice one :)

**Comment by KevinHock** [comment]
Thanks a bunch 🙏

**Pull request merged**